.github/workflows/02_scala_container_publishing.yml

```diff
@@ -91,10 +113,30 @@ jobs:
           sbom: true
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
-      
+
+      - name: Save Docker Image Locally
+        run: |
+          mkdir -p ./docker-artifacts
+          image_tag=$(echo "${{ steps.meta.outputs.tags }}" | head -n1)
+          docker pull "$image_tag"
+          docker save "$image_tag" > ./docker-artifacts/goatrodeo.oci.tar
+
+      - name: Run Grinder Scan on Docker Image
+        uses: spice-labs-inc/grinder-scan-action@main
+        with:
+          spice_pass: ${{ secrets.GRINDER_JWT }}
+          file_path: ./docker-artifacts
+
+      - name: Upload Docker Image Scan Results
+        uses: actions/upload-artifact@v4
+        with:
+          name: grinder-scan-image-${{ github.run_id }}
```

Comment:
Why do we upload to the workflow run's assets? (should we not be uploading with grinder to our own systems?)